### DISCUSSION TITLE
What is considered a "long" silence in the audio files of a dataset?

### DISCUSSION THREAD

**Discussion by dostee-hemin** [coqui-ai/TTS#2012] [General Q&A] [answered]
The speakers in my dataset use natural pauses when they talk as it is just their normal way of speaking. But I've heard my audio files should not contain "long" silences in the middle because this might confuse the model. I can say with certainty that these natural pauses are no longer than 700ms, but is this considered a long silence by the model's standards? Is it better to trim these silences or keep them so the model can understand how to pause naturally?

**Comment by erogol** [comment]
It depends but the in general, models don't like silences as it breaks the alignment between text and audio

**Comment by dostee-hemin** [comment, marked as answer]
From my own experimentation, I have discovered the following:

Even with voice samples that have no more than 550ms silence between words, the model can not understand how to correctly and naturally pause between words. In fact, training the model on voice samples like these completely ruins it's previous training and the model will never learn how to talk again.

Punctuation like commas, periods, and more can be understood by the model because they act as markers that let the model know when exactly to pause. However, we can't expect the model to look at the words of a sentence and understand where the most natural place to pause is since it has no understanding of the meaning of these words.

If you wish to add natural pauses between words in your model, then I suggest adding a specific special character between the words where you want silences so the model can see exactly where it should pause. For example, if the character "@" represents a natural pause, your sentences could be formatted to look like this:

"On an early summer morning @ in a land far far away @ the giant laid amongst the trees @ contemplating the meaning of the universe."

If anyone has a suggestion for how to automate this process, then please let me know.
